ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/user_management_service.proto

```diff
@@ -96,8 +119,25 @@ message User {
     // Ledger API clients SHOULD set this field to a non-empty value for all users to
     // enable the users to act on the ledger using their own Daml party.
     // Users for participant administrators MAY have an associated primary party.
+    //
     // Optional
+    // Modifiable
```

Comment:
I'm a bit worried about that part. The way we set things up at the moment, a lot of client code resolves this once and then caches it either in memory or maybe even worse in a database.

How do we expect clients to discover and handle changes? Same thing applies to metadata although that isn’t used by any existing clients so we at least don't break assumptions but I'm still not clear how we expect to write clients for this.

Reply:
I haven't considered implications of making 'primary_party' modifiable.
What client code are you referring to? If it's tooling in the SDK repo we can fix it. 

We can put the whole udpateUser/updatePartyDetails feature behind a feature flag if that helps.


> How do we expect clients to discover and handle changes?
Do you mean discovering that someone updated a participant user? I'd say the entity that made the change and is interested in the change would be the same entity, e.g. a participant admin user representing a participant operator.

Reply:
> What client code are you referring to? If it's tooling in the SDK repo we can fix it.

There is some stuff in the SDK, e.g., Navigator that i believe caches the primary party forever at the moment. I think overall, most of that is probably fine to keep at least for a first version. I'm more worried about client code I'm not ware.

> Do you mean discovering that someone updated a participant user? I'd say the entity that made the change and is interested in the change would be the same entity, e.g. a participant admin user representing a participant operator.

Yeah I think you're probably right that most cases are manageable fairly easily.

Reply:
> The changes look reasonable. My main question is how we expect users to actually write apps against this. At the moment, it seems like the best I can do is poll which I suppose works but feels a bit weird. I'm also particularly concerned about breaking existing apps that make assumptions about primary parties not changing.

What kind of apps and what would be the use case for polling for updates in a user resource?

(My intuition is the entity running a client app shall make sure a participant user it relies on has the correct rights, primary party, is not deactivated etc.. either by talking to another entity who has administrator rights on the participant node, or itself has the administrator rights)

Reply:
Having thought a bit more about primary party. I think I’m less worried about it. For a change that needs to have effect immediately, I expect the rights to also remove that party in which case requests will just fail. And for changes where the rights still allow to act as that party, having clients continue using the cached party seems fairly harmless.

Reply:
I would like to close this discussion. I can see that the current navigator will have a hard time to deal with the changes to the primary_party. We can solve this  by doing one of 2 things

- Claim it is unsupported scenario in navigator and document the fact that a restart is required (seems fair for a dev tool)
- Add a story to the epic to rectify it and make the navigator adjust to the shifting landscape

I am in favour of the former option

Reply:
I agree with doing the "unsupported scenario in navigator" part.